netlogo-gui/src/main/editor/EditorArea.java

```diff
@@ -101,6 +114,30 @@ public EditorArea(int rows, int columns,
     getInputMap().put
         (javax.swing.KeyStroke.getKeyStroke(java.awt.event.KeyEvent.VK_F1, 0),
             Actions.quickHelpAction(colorizer, i18n));
+
+    getInputMap().put
+            (javax.swing.KeyStroke.getKeyStroke(java.awt.event.KeyEvent.VK_SPACE,
+                    java.awt.event.InputEvent.CTRL_DOWN_MASK),
+                    Actions.codeCompletionAction(codeCompletionPopup));
+
+    if(codeCompletionPopup != null){
+      codeCompletionPopup.init(this);
+    }
+
+    getDocument().addDocumentListener(new DocumentListener() {
+      @Override
+      public void insertUpdate(DocumentEvent e) {
+        codeCompletionPopup.showPopup();
+      }
+
+      @Override
+      public void removeUpdate(DocumentEvent e) {
+        codeCompletionPopup.showPopup();
+      }
+
+      @Override
+      public void changedUpdate(DocumentEvent e) {}
+    });
```

Comment:
All of this code works, but it's a lot of detail for a particular feature of the `EditorArea`. I'm going to leave a longer comment on the PR suggesting some ways that we can add functionality here without having `EditorArea` depend directly on `CodeCompletionPopup`.